spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/web/reactive/server/WebFluxTags.java

```diff
@@ -117,4 +120,31 @@ public static Tag exception(Throwable exception) {
 		return EXCEPTION_NONE;
 	}
 
+	/**
+	 * Creates a {@code outcome} tag based on the response status of the given
+	 * {@code exchange}.
+	 * @param exchange the exchange
+	 * @return the "outcome" tag derived from the response status
+	 */
+	public static Tag outcome(ServerWebExchange exchange) {
+		if (exchange != null && exchange.getResponse().getStatusCode() != null) {
+			HttpStatus status = exchange.getResponse().getStatusCode();
+			String outcome;
+			if (status.is1xxInformational() || status.is2xxSuccessful()
+					|| status.is3xxRedirection()) {
+				outcome = "SUCCESS";
```

Comment:
These should be constant values or an enum since they are used in multiple places

Reply:
Should these outcome values be treated the same as `OUTCOME_UNKNOWN` and made into a constant value to avoid the creation for each request?

Reply:
> These should be constant values or an enum since they are used in multiple places

Perhaps. In terms of main code, they're used once on the MVC side and once on the WebFlux side. Sometimes duplication is preferable to creating an artificial construct that would allow them to be shared.

Reply:
> Should these outcome values be treated the same as OUTCOME_UNKNOWN and made into a constant value to avoid the creation for each request?

Yes, I think so. While the strings themselves aren't a problem, it would be better to avoid creating a new `Tag` each time.

@michaelmcfadyen Thank you for the pull request. Would you like to make this amendment?

Reply:
sure, will update the PR later today

Reply:
updated PR to use constants for each outcome tag